### ISSUE TITLE
Separate test and training sets to detect overfitting

### ISSUE THREAD

**Issue by jyegerlehner** [ibab/tensorflow-wavenet#134]
I guess we could divide the data into separate training and test sets, and detect overfitting in the usual way: report both training and test loss, and when the loss on the test set starts to rise while training set loss is falling, we know we are starting to overfit. I think we could be overfitting, at least in the experiments like I've done training on a single-speaker's data. 

The authors make the statement 
> Because log-likelihoods are tractable, we tune hyper-
parameters on a validation set and can easily measure if the model is overfitting or underfitting.

Not clear to me why tractability of log-likelihoods is related to detecting overfitting or underfitting. Does that mean they are using some entropy measure to detect it? Any of you probability theory experts care to enlighten us? 

Please share any ideas about how we should detect overfitting. THanks.

**Comment by ibab** [comment]
When I first read that it sounded like they just meant that they calculated the cross-entropy with respect to a validation dataset.
(As opposed to using something like a generative adversarial training procedure).
Maybe they wanted to point out that you can compare the log-likelihood values between runs in order to optimize the hyperparameters.

**Comment by jyegerlehner** [comment]
OK thanks. I see jrao1 already has an issue on the same topic.
